custom_components/better_thermostat/models/utils.py

```diff
@@ -34,13 +36,18 @@ def temperature_calibration(self):
 	# FIXME: Write docstring
 	
 	state = self.hass.states.get(self.heater_entity_id).attributes
-	new_calibration = abs(float(round((float(self._target_temp) - float(self._cur_temp)) + float(state.get('local_temperature')), 2)))
-	if new_calibration < float(self._TRV_min_temp):
-		new_calibration = float(self._TRV_min_temp)
-	if new_calibration > float(self._TRV_max_temp):
-		new_calibration = float(self._TRV_max_temp)
-	
-	return new_calibration
+	if not all([self._target_temp, self._cur_temp, state.get('local_temperature')]):
+		if not self._target_temp:
+			return float(self._TRV_min_temp)
+		return float(self._target_temp)
+	else:
+		new_calibration = abs(float(round((float(self._target_temp) - float(self._cur_temp)) + float(state.get('local_temperature')), 2)))
```

Comment:
Why is this value flipped positive when negative? I doubt that's necessary and might lead to weird values:

-7 as an example is going to be 7 so is above _TRV_min_temp and will go through.

Reply:
Please also rename this "calibration" is just wrong. It's "calibrated_target_temperature" or something like this.

Reply:
Fixed it in beta 5, at least in my local testing, i haven't a real TRV thats used this calibration type